query/control/controller_test.go

```diff
@@ -117,7 +116,7 @@ func validateUnusedMemory(t testing.TB, reg *prometheus.Registry, c control.Conf
 }
 
 func TestController_QuerySuccess(t *testing.T) {
-	ctrl, err := control.New(config)
+	ctrl, err := control.New(config, nil, executorDependencies)
```

Comment:
Instead of `nil` for these, you could use `zaptest.Logger(t)`. That way we'll get the log info on failures.